### PR TITLE
Fix broken links in documentation

### PR DESCRIPTION
**Pull request by berquist**
- Add Shiv to developer's list
- Fix some leftover old version mentions
- Fix broken links using Sphinx's `-b linkcheck`
- HTTP -> HTTPS, fix redirects

### CODE REVIEW

**Comment by berquist** [comment]
A bunch of GAMESS-UK and Jaguar links are broken...

File: doc/sphinx/data_notes.rst [hunk 1]
```diff
@@ -364,7 +364,7 @@ A list of rank 1 arrays containing the molecular orbital energies in eV. The lis
 
 **Jaguar**: the first ten virtual orbitals are printed by default. In order to print more, use the ``ipvirt`` keyword, with ``ipvirt=-1`` printing all virtual orbitals.
 
-.. _`FORMAT HIGH`: http://www.cfs.dl.ac.uk/docs/html/part3/node8.html#SECTION00083000000000000000
+.. _`FORMAT HIGH`: https://web.archive.org/web/20190809100513/http://www.cfs.dl.ac.uk/docs/html/part3/node8.html
```
Comment (shivupa) [inline doc/sphinx/data_notes.rst:367]: lol

File: doc/sphinx/how_to_install.rst [hunk 2]
```diff
@@ -16,7 +16,7 @@ Python is an open-source programming language available from https://www.python.
 
     apt-get install python python-dev
 
-NumPy (Numerical Python) adds a fast array facility and linear algebra routines to Python. It is available from https://www.numpy.org. Windows users should use the most recent NumPy installation for the Python version they have (e.g. numpy-1.0.3.1.win32-py2.4.exe for Python 2.4). Linux users are recommended to find a binary package for their distribution rather than compiling it themselves. In Debian/Ubuntu it is installed as follows (as root):
+NumPy (Numerical Python) adds a fast array facility and linear algebra routines to Python. It is available from https://numpy.org/. Windows users should use the most recent NumPy installation for the Python version they have (e.g. numpy-1.0.3.1.win32-py2.4.exe for Python 2.4). Linux users are recommended to find a binary package for their distribution rather than compiling it themselves. In Debian/Ubuntu it is installed as follows (as root):
```
Comment (shivupa) [inline doc/sphinx/how_to_install.rst:19]: Windows users should consider using a conda installation to install Python and the most recent NumPy installation.
https://docs.conda.io/projects/conda/en/latest/user-guide/install/windows.html

**Review by shivupa — APPROVED** [review]
Thanks I just have 1 minor comment (which wasn't the focus of this PR) but we could update the installation recommendation for Numpy on Windows. Let me know what you think

**Comment by shivupa** [comment]
The changes were out of scope of this PR

**Pull request merged**